catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java

```diff
@@ -311,7 +328,36 @@ public AttributeFormat getAttributeFormat() {
                 false /* tokenized */,
                 true /* multivalued */,
                 STRING_TYPE));
-        return descriptors;
+        DESCRIPTORS = Collections.unmodifiableSet(descriptors);
+
+        BASIC_METACARD = new MetacardTypeImpl(MetacardType.DEFAULT_METACARD_TYPE_NAME,
```

Comment:
My apologies Ugly was not the correct word misleading is more accurate. If the accessor methods were static I would agree with you but our hands our tied in that regard so they belong to this instance and new implies a cost the underling descriptors could be created every time and then any modification would be local and it would be perfectly OO.